sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -357,3 +361,45 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_missing_indicator():
+    X1 = np.array([
+         [-1,  -1,   1,   3],
+         [4,  -1,   0,  -1],
+         [8,  -1,   1,  0],
+         [0,  -1,   0,  15],
+         [16,  -1,   1,  19]
+    ])
+    X2 = np.array([
+         [5,  1,   -1,   -1],
+         [-1,  -1,   2,  3],
+         [2,  3,   4,  0],
+         [0,  -1,   5,  -1],
+         [11,  -1,   1,  1]
+    ])
+
```

Comment:
Could you explain what ``retype`` means. I guess it it return type but does it mean it should call ``MI.fit(retype(X1))`` and ``MI.transform(retype(X2))`` ?